fruit/fruit.js

```diff
@@ -50,7 +411,51 @@ export default () => {
       /* activateCb = e => {
         console.log('activate', e);
       }; */
+
+      const _getActionFrameIndex = (f, frameTimes) => {
+        let i;
+        for (i = 0; i < frameTimes.length; i++) {
+          if (f >= frameTimes[i]) {
+            continue;
+          } else {
+            break;
+          }
+        }
+        return i;
+      };
+
+      const eatFrameIndices = [500, 800, 1100];
+      let lastEatFrameIndex = -1;
+      let particleAlreadyInScene= false;
+      let removeFruitFromApp = false;
+      let circlePlay = false;
+      let materialStartTime = -1;
+      let dir = new THREE.Vector3();
+      let storeMaterial = false;
+      let materials=[];
       frameCb = (timestamp, timeDiff) => {
+
+        //#################### store avatar material ############################
+        if(localPlayer.avatar && !storeMaterial){
```

Comment:
This looks great, but we should put this in the `app`, not the fruits.

There is already similar code for the powerup hair here: https://github.com/webaverse/app/blob/376df93433b51ef41587ee44c75e8fd8bde5f29f/character-fx.js#L179

Reply:
Probably best to make this an action on the player (like `type: 'cure'`) and let the character fx code handle it.